Kernel/Arch/aarch64/RegisterState.h

```diff
@@ -11,35 +11,124 @@
 #include <AK/Platform.h>
 VALIDATE_IS_AARCH64()
 
+#include <AK/Types.h>
+#include <Kernel/Arch/CPU.h>
+
 namespace Kernel {
 
 struct RegisterState {
-    u64 x[31];     // Saved general purpose registers
+    u64 x[32];     // Saved general purpose registers
     u64 spsr_el1;  // Save Processor Status Register, EL1
     u64 elr_el1;   // Exception Link Register, EL1
     u64 tpidr_el1; // EL0 thread ID
     u64 sp_el0;    // EL0 stack pointer
 
-    FlatPtr userspace_sp() const { return 0; }
+    FlatPtr pc;
+    FlatPtr nzcv;
+
+    FlatPtr userspace_sp() const { return sp_el0; }
     void set_userspace_sp(FlatPtr value)
     {
-        (void)value;
-        TODO_AARCH64();
+        sp_el0 = value;
     }
-    FlatPtr ip() const { return 0; }
+    FlatPtr ip() const { return pc; }
     void set_ip(FlatPtr value)
     {
-        (void)value;
-        TODO_AARCH64();
+        pc = value;
+    }
+    FlatPtr bp() const { return x[29]; }
+    void set_bp(FlatPtr value) { x[29] = value; }
+    FlatPtr flags() const { return nzcv; }
+    void set_flags(FlatPtr value) { nzcv = value; }
+    void set_return_reg(FlatPtr value) { x[0] = value; }
+
+    void capture_syscall_params(FlatPtr& function, FlatPtr& arg1, FlatPtr& arg2, FlatPtr& arg3, FlatPtr& arg4) const
+    {
+        function = x[8];
+        arg1 = x[1];
+        arg2 = x[2];
+        arg3 = x[3];
+        arg4 = x[4];
     }
-    FlatPtr bp() const { TODO_AARCH64(); }
 };
 
 inline void copy_kernel_registers_into_ptrace_registers(PtraceRegisters& ptrace_regs, RegisterState const& kernel_regs)
```

Comment:
The implementation of `copy_kernel_registers_into_ptrace_registers` and the above changes to add implementations of RegisterState's methods are independent and should be in separate commits.

kernel regs --> ptrace regs and ptrace regs --> kernel regs can be in the same commit though

Reply:
i understand that each commit should yield in coherent (i.e. runnable) state, right?